### DISCUSSION TITLE
Best way to debug a plugin

### DISCUSSION THREAD

**Discussion by SpyrexDE** [SteamDeckHomebrew/decky-loader#10] [Q&A] [answered]
What is the best way to debug a plugin at this time? Will there ever be a good development workflow with some kind of live reload?

**Comment by WerWolv** [comment]
Automatic reload is being worked on right now by @marios8543. 
The way I did it was by connecting to my steam deck over ssh directly with vscode and edit the plugin file there and just manually execute the loader in the console there. That way you have log output from python.
If you want log output from the HTML / JS part too, you can connect to `http://<steam_deck_ip>:8081` and click on QuickAccess there

**Reply by SpyrexDE** [reply]
Sounds great but for me running loader.py does nothing.

**Reply by marios8543** [reply]
You're supposed to run main.py

**Reply by SpyrexDE** [reply]
Well, that gives me a bunch of errors(`('127.0.0.1', 1337): address already in use
`), redraws the plugins tab but not with the updated data.

**Comment by marios8543** [comment]
> also you can reload plugins by running `curl http://localhost:1337/plugins/reload`

**Reply by SpyrexDE** [reply]
Doesn't work for me.

**Reply by marios8543** [reply]
Live reloading now implemented
https://github.com/SteamDeckHomebrew/PluginLoader/commit/8c142c01bda31a164cf5bacf5a7ae85366334a61

**Reply by SpyrexDE** [reply]
You still need to reload the iframe, maybe automate that as well...
Additionally, it does not remove the previously loaded plugins.

**Reply by marios8543** [reply]
> You still need to reload the iframe, maybe automate that as well...
done https://github.com/SteamDeckHomebrew/PluginLoader/commit/8d0fe5c45ac67ea4eef9a7c6a5e4eaaa57ad01d3

**Comment by TrainDoctor** [comment, marked as answer]
If you are using a chromium browser (I recommend ungoogled-chromium), ``chrome://inspect/`` and then adding the ip address of your deck and port 8081 will get you a more up to date set of tools to use, but not as nicely integrated into the browser as ``ip.ad.dre.ss:8081``.
